seed/java-sdk/audiences/src/main/java/com/seed/audiences/resources/foo/FooClient.java

```diff
@@ -70,7 +70,8 @@ public ImportingType find(FindRequest request, RequestOptions requestOptions) {
             }
             throw new ApiError(
                     response.code(),
-                    ObjectMappers.JSON_MAPPER.readValue(response.body().string(), Object.class));
+                    ObjectMappers.JSON_MAPPER.readValue(
+                            response.body() != null ? response.body().toString() : "{}", Object.class));
```

Comment:
```java
Response response = client.newCall(okhttpRequest).execute();
String text = response.body().string();
if (response.isSuccessful()) {
  return ObjectMappers.JSON_MAPPER.readValue(text, ImportingType.class);
}
throw new ApiError(
   response.code(),
   text == null ? "" : text)
```

Reply:
@dcb6  what do you think about getting the text into a variable before hand -- this will set us up in a better place to add custom thrown errors to java imo and handle json decode errors as well. 

Stealing from [TypScript](https://github.com/fern-api/fern/blob/91deacfd43be76321340482b52774e2cfbd71301/generators/typescript/utils/core-utilities/fetcher/src/fetcher/Fetcher.ts#L137-L149)

Reply:
i think the NPE is actually on the `response.body().string()` call itself -- `response.body()` itself can bell null, intellij is even warning me about it:

![CleanShot 2024-05-07 at 09 55 58](https://github.com/fern-api/fern/assets/20914602/41e3834d-cde6-4492-a9ab-d3ead71e1a93)

so i could make a variable out of response.body() but that's about it

(also noted that i erroneously changed it to `toString()`, will fix)

Reply:
i see what you're pointing out though:
![CleanShot 2024-05-07 at 12 41 04](https://github.com/fern-api/fern/assets/107228315/a41e69e4-3752-4242-81cc-e99484045091)

how about this:
```
            Response response = client.newCall(okhttpRequest).execute();
            ResponseBody responseBody = response.body();
            String responseBodyString = responseBody != null ? responseBody.toString() : "{}";
            if (response.isSuccessful()) {
                return ObjectMappers.JSON_MAPPER.readValue(responseBodyString, ImportingType.class);
            }
            throw new ApiError(
                    response.code(),
                    ObjectMappers.JSON_MAPPER.readValue(responseBodyString, Object.class));
```